### ISSUE TITLE
Sent response body cannot be easily accessible in continuing middlewares

### ISSUE THREAD

**Issue by esteban-filardi** [oauthjs/express-oauth-server#87]
The requirement for my app is the following:

_"Log all incoming requests (including the request's response body)"._

Even if there is the option _continueMiddleware_ to continue executing a following middleware, it isn't possible to access the response of the _token()_ method in that next middleware. Because of that, I can't easily log the response of the API method to obtain a token.

Maybe it would be feasible to store the response body inside the _res_ object before send it in order to be able to have access it in the following middlewares of the chain.

Alternatively, an option could be added to just obtaining the response body of _oauth-node-server_ but not sending it over automatically.

**Comment by AakashPat** [comment]
I am looking for something similar. 
The token() method returns an error in the following format in case a wrong set of credentials is used.
`{
    "error": "invalid_grant",
    "error_description": "Invalid grant: user credentials are invalid"
}`

What I intend to do is to return an appropriate _error_description_ depending on the reason for grant failure. Example: User account does not exist. User account is not verified. Incorrect password.

Does anyone have any idea of how I could achieve this?

**Comment by nkanand4** [comment]
@esteban-filardi you can access response of the `token()` method by accessing it from the response which will be available to you in the next handler, `res.locals.oauth.token`. Refer https://github.com/oauthjs/express-oauth-server/blob/master/index.js#L116

**Comment by mdavies-solsys** [comment]
I have to agree @esteban-filardi here. While @nkanand4 is correct in that both the token and authorize handler provide the token/code in the `res.locals`, using continueMiddleware for simple overrides has left me building a response that has already been built. In fact, I only want to override the token handler but now I must rebuild the response for the authorize as well. Hopefully I'm not missing something here because I'd like to make a PR!
